cfme/storage/volume_snapshot.py

```diff
@@ -0,0 +1,235 @@
+# -*- coding: utf-8 -*-
+import attr
+
+from navmazing import NavigateToSibling, NavigateToAttribute
+from widgetastic_manageiq import (
+    Accordion,
+    BaseEntitiesView,
+    BreadCrumb,
+    ItemsToolBarViewSelector,
+    ManageIQTree,
+    SummaryTable
+)
+from widgetastic_patternfly import Button, Dropdown
+from widgetastic.widget import View, Text
+
+from cfme.base.ui import BaseLoggedInPage
+from cfme.common import TagPageView, WidgetasticTaggable
+from cfme.exceptions import SnapshotNotFoundError, ItemNotFound
+from cfme.modeling.base import BaseCollection, BaseEntity
+from cfme.utils.appliance.implementations.ui import CFMENavigateStep, navigator, navigate_to
+from cfme.utils.wait import wait_for
+
+
+class VolumeSnapshotToolbar(View):
+    """The toolbar on the Volume Snapshot page"""
+    policy = Dropdown('Policy')
+    download = Dropdown('Download')
+    view_selector = View.nested(ItemsToolBarViewSelector)
+
+
+class VolumeSnapshotDetailsToolbar(View):
+    """The toolbar on the Volume Snapshot detail page"""
+    configuration = Dropdown('Configuration')
+    policy = Dropdown('Policy')
+    download = Button('Download summary in PDF format')
+
+
+class VolumeSnapshotDetailsEntities(View):
+    """The entities on the Volume Snapshot detail page"""
+    breadcrumb = BreadCrumb()
+    properties = SummaryTable('Properties')
+    relationships = SummaryTable('Relationships')
+    smart_management = SummaryTable('Smart Management')
+
+
+class VolumeSnapshotDetailSidebar(View):
+    """The accordion on the Volume Snapshot details page"""
+    @View.nested
+    class properties(Accordion):  # noqa
+        tree = ManageIQTree()
+
+    @View.nested
+    class relationships(Accordion):  # noqa
+        tree = ManageIQTree()
+
+
+class VolumeSnapshotView(BaseLoggedInPage):
+    """A base view for all the Volume Snapshot pages"""
+    title = Text('.//div[@id="center_div" or @id="main-content"]//h1')
```

Comment:
Because of commits previous discussion has been hidden.

@digitronik raised a good point about title definition having to be repeated.

BaseEntitiesView includes a title widget via its conditionalview:
https://github.com/ManageIQ/integration_tests/blob/master/widgetastic_manageiq/__init__.py#L2819

So I think you can drop title from this generic top-level view. There's no need to define it in `VolumeSnapshotAllView` because `BaseEntitiesView` is included there. There is a need to define a `title` widget for the `VolumeSnapshotDetailsEntities`, as this is nested in `VolumeSnapshotDetailsView`.

This will provide a consistent path to the title widget via `view.entities.title` - regardless of what type the `view` object is.

Reply:
right @mshriver; going to add changes waiting for #6085. Due to this snapshot failing on PRT.